addons/mrp/models/mrp_production.py

```diff
@@ -527,7 +554,8 @@ def action_cancel(self):
             finish_moves = production.move_finished_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
             raw_moves = production.move_raw_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
             (finish_moves | raw_moves)._action_cancel()
-
+            picking_ids = production.picking_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
+            picking_ids.action_cancel()
```

Comment:
i thought the propagate_cancel on the pull rules should trigger the cancel on the next/previous moves. why is it done here?

Reply:
(propogate_cancel ) It will not cancel previous moves but cancel then next moves thats why we have done this ...

Reply:
ok. really not sure that we should implement this logic in mrp. mrp should use the stock concepts, else we'll have te re-write stock in mrp as in v10 :/

let me check a little. i'll go back to you